src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/AnalysisBasedMetadataManager.cs

```diff
@@ -47,6 +48,7 @@ public AnalysisBasedMetadataManager(CompilerTypeSystemContext typeSystemContext)
             StackTraceEmissionPolicy stackTracePolicy,
             DynamicInvokeThunkGenerationPolicy invokeThunkGenerationPolicy,
             IEnumerable<ModuleDesc> modulesWithMetadata,
+            IEnumerable<TypeDesc> forcedTypes,
             IEnumerable<ReflectableEntity<TypeDesc>> reflectableTypes,
```

Comment:
What's the difference between `forcedTypes` and `reflectableTypes` - per the change description these should be basically identical now...

Reply:
`reflectableTypes` are types that are eligible to be reflection visible, but are not rooted. `forcedTypes` are rooted.

I forgot to mention that we have a concept of reflection blocked types - types that produce a special `System.Type` at runtime.

`reflectableTypes` basically helps compilation phase to decide whether something should be blocked or not.

It unfortunately doesn't match well with the below `reflectableFields` and `reflectableMethods` because types have the unfortunate property of becoming visible to reflection just because they exist (unless blocked).